pkg/pluginmanager/manager.go

```diff
@@ -681,10 +691,19 @@ func DeletePlugin(options DeletePluginOptions) error {
 	return nil
 }
 
-// SyncPlugins automatically downloads all available plugins to users machine
+// SyncPlugins automatically installs all plugins of the current contexts
 func SyncPlugins() error {
 	log.Info("Checking for required plugins...")
-	plugins, err := AvailablePlugins()
+	var plugins []discovery.Discovered
+	var err error
+	if configlib.IsFeatureActivated(constants.FeatureCentralRepository) {
+		plugins, err = DiscoverServerPlugins()
```

Comment:
I understand the rationale behind this, but it might be good to put a comment to explain why.
Also, this being a rather significant user-facing change. Please make sure to summarize it in a sentence or two under "Release Notes" part of the PR since at some point we will turn on the tooling to aggregate release notes in our releases.

Reply:
Done